src/sentry/tasks/derive_code_mappings.py

```diff
@@ -74,11 +74,17 @@ def derive_code_mappings(
         with lock.acquire():
             trees = installation.get_trees_for_org()
     except ApiError as error:
-        json_data: JSONData = error.json
-        msg: str = json_data.get("message")
+        msg = error.text
+        if error.json:
+            json_data: JSONData = error.json
+            msg = json_data.get("message")
+        extra["error"] = msg
+
         if msg == "Not Found":
-            logger.warning("The org has uninstalled the Sentry App.")
+            logger.warning("The org has uninstalled the Sentry App.", extra=extra)
             return
+
+        raise error  # Let's be report the issue
```

Comment:
When I wrote https://github.com/getsentry/sentry/pull/42487, I introduced this regression. We need to report all API Errors.